src/ngMock/angular-mocks.js

```diff
@@ -556,9 +556,16 @@ angular.mock.$IntervalProvider = function() {
      */
     $interval.flush = function(millis) {
       now += millis;
-      while (repeatFns.length && repeatFns[0].nextTime <= now) {
+      while (repeatFns.length && (repeatFns[0].nextTime <= now || !repeatFns[0].nextTime)) {
         var task = repeatFns[0];
         task.fn();
+        // when a delay of zero is given
+        // it has to be replaced with one
+        // to avoid running the while loop
+        // one more time than the specified
+        if (!task.delay) {
+          task.delay = task.nextTime = 1;
```

Comment:
Nice! I wasn't sure about changing `$interval` but I like this implementation. Thanks!